Update doc - tests - and conf template to reflect logic around recover_after_data_nodes (#13415)
crate/crate: docs/admin/system-information.rst

The check for the
:ref:`gateway.expected_data_nodes <gateway.expected_data_nodes>` setting checks

> if the setting match the actual number of data nodes present in the cluster.

Reviewer: ```suggestion
if the setting matches the actual number of data nodes present in the cluster.
```

> The value of this settings is checked against 

Reviewer: ```suggestion
The value of this setting is checked against 
```

:ref:`gateway.recover_after_data_nodes
<gateway.recover_after_data_nodes>` (or 
:ref:`gateway.recover_after_nodes <gateway.recover_after_nodes>`

> ) during startup to decide if the cluster can proceed with 

Reviewer: I think the closing bracket needs to be on the previous line, otherwise, this renders an additional space:
<img width="831" alt="Screenshot 2022-12-30 at 14 31 57" src="https://user-images.githubusercontent.com/218003/210075594-0014bd86-817f-4c2e-9246-8e83d03b2de0.png">

:ref:`recovery <gloss-shard-recovery>`.

> If the actual number of nodes is below the number expected this is flagged to

Reviewer: ```suggestion
If the actual number of nodes is below the expected number of nodes, this is flagged to
```

> indicate some node[s] are down. If the actual number is greater this is 

Reviewer: I think you can drop the "[s]", otherwise you would also need to use "is/are" later in the sentence.
```suggestion
indicate some nodes are down. If the actual number is greater, this is 
```
